### PR TITLE
Added design for TDEngine integration

### PR DESCRIPTION
**Pull request by BtXin**
**What this PR does / why we need it**:
Design for integrate TDEngine
**Will this PR make the community happier**? 
Yes
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**How is this PR tested**
- [ ] unit test
- [ ] e2e test
- [x] other (please specify)
Design doc

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Review by tomqin93 — COMMENTED** [review]
@BtXin added 2 comments, rest looks good

File: docs/design/shifu-framework/telemetry-service-TDEngine.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+type TelemetryRequest struct {
+	RawData              []byte                `json:"rawData,omitempty"`
+	httpAddress          string                `json:"httpAddress,omitempty"`
+	MQTTSetting          *MQTTSetting          `json:"mqttSetting,omitempty"`
+	SQLConnectionSetting *SQLConnectionSetting `json:"sqlConnectionSetting,omitempty"`
+}
+```
+
+We use the name `SQLConnection` instead of `TDEngineConnection` because we want to it to be a generic DB connection setting for scalability concerns. We want to use it to conenct to various DBs instead of TDEngine alone.
```
[the remark this answers is not in the document]
Comment (BtXin) [reply docs/design/shifu-framework/telemetry-service-TDEngine.md:55]: Make sense, added it to sqlconnection struct.

File: docs/design/shifu-framework/telemetry-service-TDEngine.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+We also will add the corresponding connection setting as a go struct and will add it to `TelemetryRequest`
+
+```go
+type SQLConnectionSetting struct {
+	DBServerAddress *string `json:"db_server_address,omitempty"`
+	DBUserName      *string `json:"db_username,omitempty"`
+	DBSecret        *string `json:"db_secret,omitempty"`
+	DBName          *string `json:"db_name,omitempty"`
+	DBTable         *string `json:"db_table,omitempty"`
+	DBType          *string `json:"db_type,omitempty"`
```
Comment (tomqin93) [inline docs/design/shifu-framework/telemetry-service-TDEngine.md:44]: Let's use type instead of literal string

**Review by tomqin93 — COMMENTED** [review]
@BtXin added 1 minor comment

**Pull request merged**